### PR TITLE
Fix and clarify VAV minimum flow sizing, especially when VRP calculations adjust the min flow

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
Addresses #6055.

1. When the ventilation rate procedure (VRP) determines that the minimum zone flow rate needs to be increased, the zone maximum flow rate is also increased if it was less than the new minimum.
2. The Std 62.1 sizing report now shows the updated values.
3. ~In the Std 62.1 sizing report, all columns which report volume flow rate now use 4 figures after the decimal (previously some had only 2 or 3 causing equal values to appear different).~ *Removed this change for now to reduce table diffs.*
4. VAV mininimum flow autosizing now *always* uses FinalZoneSizing( CurZoneEqNum ).DesCoolVolFlowMin to set the minimum flow rate or minimum flow fraction if a sizing run has been done.  This variable holds the maximum flow derived from 
  a. the design outdoor air flow (including VRP adjustments) and 
  b. the Sizing:Zone input fields "Cooling Minimum Air Flow per Zone Floor Area", 
  c. "Cooling Minimum Air Flow", and
  d. "Cooling Minimum Air Flow Fraction".
5. Previously, if Sizing:Zone "Cooling Minimum Air Flow Fraction" was not blank, this fraction would be used to size the VAV minimum flow rate, regardless of all other inputs.  This is no longer the case. 
6. Docs and IDD notes have been updated to clarify what values impact the VAV minimum flow autosizing. 

### Remaining questions ###
 - [ ] Does the same check on minimum ventilation efficiency need to be done for heating?  Currently this is only checked for cooling.
 - [ ] Need more unit tests and/or test files to look at this for a simple, 3-zone system over a wide range of cases.
 - [ ] Need more unit tests and/or test files to check if zones with recirculation are being handled correctly.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by nrel-bot-3** [comment]
@mjwitte @lgentile it has been 16 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@mjwitte @lgentile it has been 15 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by mjwitte** [comment]
eio/err/table diffs are from terminal units where previously the Design Size Constant Minimum Air Flow Fraction was zero, and now it is not.  Not change in results though, because most of these had a hard value for this input.
```
    **   ~~~   ** User-Specified Minimum Cooling Air Flow Fraction of 1.00000 [m3/s]
-   **   ~~~   ** differs from Design Size Minimum Cooling Air Flow Fraction of 0.00000 [m3/s]
+   **   ~~~   ** differs from Design Size Minimum Cooling Air Flow Fraction of 0.72766 [m3/s]

  Component Sizing Information, AirTerminal:SingleDuct:VAV:NoReheat, SPACE1-1 DOAS AIR TERMINAL, Design Size Maximum Air Flow Rate [m3/s], 0.10384
- Component Sizing Information, AirTerminal:SingleDuct:VAV:NoReheat, SPACE1-1 DOAS AIR TERMINAL, Design Size Constant Minimum Air Flow Fraction, 0.00000
+ Component Sizing Information, AirTerminal:SingleDuct:VAV:NoReheat, SPACE1-1 DOAS AIR TERMINAL, Design Size Constant Minimum Air Flow Fraction, 0.72766
  Component Sizing Information, AirTerminal:SingleDuct:VAV:NoReheat, SPACE1-1 DOAS AIR TERMINAL, User-Specified Constant Minimum Air Flow Fraction, 1.00000
```
Re-running linux tests - strange errors before.  Otherwise, this is ready for review.

File: doc/input-output-reference/src/overview/group-air-distribution-equipment.tex [hunk 1]
```diff
@@ -192,13 +192,13 @@ \subsubsection{Inputs}\label{inputs-2-001}
...
 
-This field is autosizable and defaulted to autosize. The values for autosizing are picked up from the Sizing:Zone input fields "Cooling Minimum Air Flow per Zone Floor Area", "Cooling Minimum Air Flow", and "Cooling Minimum Air Flow Fraction". If there is no sizing calculation a default of .000762 cubic meters per second per square meter of zone floor area (0.15 cfm/ft2) is used.
+This field is autosizable and defaulted to autosize. The autosized flow fraction is calculated using the maximum flow rate derived from the design outdoor air flow (including VRP adjustments) and the Sizing:Zone input fields "Cooling Minimum Air Flow per Zone Floor Area", "Cooling Minimum Air Flow", and "Cooling Minimum Air Flow Fraction". If there is no sizing calculation the defaults of 0.000762 cubic meters per second per square meter of zone floor area (0.15 cfm/ft2) and 0.2 are used.
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-air-distribution-equipment.tex:195]: I'm going through this carefully so there may be many comments.  I'll recap any major things in the final review comment.

File: doc/input-output-reference/src/overview/group-design-objects.tex [hunk 2]
```diff
@@ -1605,7 +1605,7 @@ \subsubsection{Inputs}\label{inputs-6-006}
 
 \paragraph{Field: System Outdoor Air Method}\label{field-system-outdoor-air-method-000}
 
-The method used to calculate the system minimum outdoor air flow. The two choices are ZoneSum and VentilationRateProcedure (VRP). ZoneSum sums the outdoor air flows across all zones served by the system. VRP uses the multi-zone equations defined in 62.1-2007 to calculate the system outdoor air flow. VRP considers zone air distribution effectiveness and zone diversification of outdoor air fractions.
+The method used to calculate the system minimum outdoor air flow. The two choices are ZoneSum and VentilationRateProcedure (VRP). ZoneSum sums the outdoor air flows across all zones served by the system. VRP uses the multi-zone equations defined in 62.1-2007 to calculate the system outdoor air flow. VRP considers zone air distribution effectiveness and zone diversification of outdoor air fractions. VRP may also adjust autosized air terminal maximum and minimum supply flow rates if needed to ensure adequate outdoor air flow rate to each zone.
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-design-objects.tex:1608]: Doc changes look good.  Built them locally and changes are fine.

File: idd/Energy+.idd.in [hunk 3]
```diff
@@ -31904,7 +31904,7 @@ Sizing:Zone,
       \type real
       \minimum 0
       \default 0.2
-      \note This input is currently used in sizing the Fan minimum Flow Rate.
+      \note This input is currently used in sizing the VAV air terminal unit and fan minimum flow rate
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in:31907]: :+1:

File: src/EnergyPlus/DataSizing.hh [hunk 4]
```diff
@@ -326,7 +326,6 @@ namespace DataSizing {
 		Real64 DesCoolMinAirFlowPerArea; // design cooling minimum air flow rate per zone area [m3/s / m2]
 		Real64 DesCoolMinAirFlow; // design cooling minimum air flow rate [m3/s]
 		Real64 DesCoolMinAirFlowFrac; // design cooling minimum air flow rate fraction
-		bool DesCoolMinAirFlowFracUsInpFlg; // user input for minimum air flow rate fraction
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataSizing.hh:329]: Interesting seeing a variable removed.  I like it, just not very common it seems.  I'll give 200 internet points for every line of code removed.

File: src/EnergyPlus/SimAirServingZones.cc [hunk 5]
```diff
@@ -4684,7 +4685,7 @@ namespace SimAirServingZones {
 		int MatchingCooledZoneNum; // temporary variable
 		Real64 termunitsizingtempfrac; // 1.0/(1.0+termunitsizing(ctrlzone)%inducrat)
 		Real64 termunitsizingtemp; // (1.0+termunitsizing(ctrlzone)%inducrat)
-		Real64 VozClg( 0.0 ); // corrected (for ventilation efficiency) zone outside air flaw rate [m3/s]
```
Comment (Myoldmopar) [inline src/EnergyPlus/SimAirServingZones.cc:4687]: :+1: for removing the **flaw**

File: src/EnergyPlus/SimAirServingZones.cc [hunk 6]
```diff
@@ -6682,8 +6686,28 @@ namespace SimAirServingZones {
 			FinalZoneSizing( CtrlZoneNum ).ZpzClgByZone = ZoneOAFrac;
 			// save new (increased) minimum flow rate
 			FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin = AvailSAFlow;
+			// make sure the max flow rate is >= the new minimum flow rate
+			FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlow = max( AvailSAFlow, FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlow );
```
Comment (Myoldmopar) [inline src/EnergyPlus/SimAirServingZones.cc:6690]: OK, so you calculate a new value of `DesCoolVolFlow` here.  Then it is used a few lines below in calculating a value for the local variables `VpzClgByZone` and `VdzClgByZone`.  And those are reported.  Is `DesCoolVolFlow` supposed to be used anywhere else?  I don't see it in the other places below.  But I'll keep looking.

File: src/EnergyPlus/SingleDuct.cc [hunk 7]
```diff
@@ -2061,21 +2050,21 @@ namespace SingleDuct {
...
 			else {
-				// no zone sizing values available; use max of min frac = 0.2 and 0.000762 [m3/s-m2]
-				FixedMinAirDes = max( 0.2*Sys( SysNum ).MaxAirVolFlowRate, 0.000762 * Zone( ZoneNum ).FloorArea * Zone( ZoneNum ).Multiplier *
-					Zone( ZoneNum ).ListMultiplier );
+				// if no zone sizing values available; use max of min frac = 0.2 and 0.000762 [m3/s-m2]
+				if ( Sys( SysNum ).MaxAirVolFlowRate > 0.0 ) {
+					FixedMinAirDes = max( 0.2 * Sys( SysNum ).MaxAirVolFlowRate, 0.000762 * Zone( ZoneNum ).FloorArea * Zone( ZoneNum ).Multiplier * Zone( ZoneNum ).ListMultiplier );
+				} else {
+					MinAirFlowFracDes = 0.0;
+				}
```
Comment (Myoldmopar) [inline src/EnergyPlus/SingleDuct.cc:2067]: Yeah I was expecting to see the `DesCoolVolFlow` used somewhere here in this file somewhere, but it's not.  I'll keep rigoring over it to see what I am not grokking.

**Review by Myoldmopar — COMMENTED** [review]
Not a yes/no so far, just ending up being comments.

File: src/EnergyPlus/SingleDuct.cc (hunk 7 above)
Comment (mjwitte) [reply src/EnergyPlus/SingleDuct.cc:2067]: It's used [here](https://github.com/NREL/EnergyPlus/blob/6259aeca0d3a457e150606567b0197209c527193/src/EnergyPlus/SingleDuct.cc), as "always" but now it's more correct for situations where the required min flow is greater than the design flow based on load.

Comment (rraustad) [reply src/EnergyPlus/SingleDuct.cc:2067]: It's OK that `DesCoolVolFlow` is not used anywhere else as long as it's the correct value where someone in the future can use it as needed.

File: doc/input-output-reference/src/overview/group-air-distribution-equipment.tex (hunk 1 above)
Comment (rraustad) [reply doc/input-output-reference/src/overview/group-air-distribution-equipment.tex:195]: This is one of those where there is no way to know if these changes are correct without a review of several input file results. I assume that was done. Any results to share? For example a list of tables that show changes in flow rates based on changes to inputs.

**Comment by mjwitte** [comment]
@rraustad Defect files added (soon) to DevSupport...\6055.  Here are results for 0.6, 0.7, and 0.8 zone Ev.  With 0.7 the Vpz-min increases, but no change in Vpz and Vdz.  With 0.8 they all change for the zone in the last row of the screen shots.
image
image
image

**Comment by mjwitte** [comment]
Just to be clear, these are results after this fix.

**Comment by rraustad** [comment]
OK, this makes sense. Why in the second case does Ev = 0.81 for the first zone (Corner_Class_1) instead of 0.7 ? @larryscheier please review these results and comment.

**Comment by rraustad** [comment]
It's Vpz-min that is changing, up to the point where Vpz-min increases the zone primary air flow rate Vpz (so that Vpz >= Vpz-min). So the minimum zone air flow rate increases as ventilation efficiency Ev increases, does that make sense? Larry, if the ventilation efficiency increases shouldn't the zone need less OA? and therefore the zone minimum air flow rate Vpz-min should decrease?

**Comment by rraustad** [comment]
This is from ASHRAE Std 62 - 2001, but the trend should still be valid. Doesn't this show that flow decreases as Ev increases?

ev

zp

**Comment by mjwitte** [comment]
Zp is the OA fraction, so higher primary flow will reduce Zp and increase Ev.  Regarding the 0.7 case above, that idf had another change as well.  New apples-to-apples idf added to the defect files folder and the screenshot [above](https://github.com/NREL/EnergyPlus/pull/6140#issuecomment-321426533) has been replaced.

**Comment by larryscheier** [comment]
1) RR: "Why in the second case does Ev = 0.81 for the first zone (Corner_Class_1) instead of 0.7?" 
LS: I don't understand this either. Input error?
2) RR:  "Larry, if the ventilation efficiency increases shouldn't the zone need less OA? and therefore the zone minimum air flow rate Vpz-min should decrease?"
LS: We decrease the system Vot (=Vou/Evs) by increasing the overall system efficiency (Evs=Minimum zone Evz) by decreasing the critical zone Zpz which is accomplished by increasing the corresponding zone's Vpz-min. When we fix the zone Evz, we then back-calculate what the required zone ventilation fraction Zd should be. As the Evz increases, the required Zpz (=Voz/Vpzmin) decreases because we have forced a larger Vpzmin:
  Evz = 1 + Xs - Zpz 
so
  Zpz = 1 + Xs - Evz  and Zpz = Voz / Vpzmin 
Solving for Vpzmin 
  Vpzmin = Voz  / Zpz

Using Multi Class 1:  (In the displaced example the system Xs = 0.546)
Zd@(Evz=0.6)  = 1 + 0.546 - 0.6 = 0.946
Vpzmin@(Zd=0.946) = Voz  / Zpz = 0.9086/0.946 = 0.9605 m3/s
Zd@(Evz=0.7)  = 1 + 0.546 - 0.7 = 0.846 
Vpzmin@(Zd=0.846) = 0.9086/0.846 = 1.074 m3/s
Zd@(Evz=0.8)  = 1 + 0.546 - 0.8 = 0.746 
Vpzmin@(Zd=0.746) = 0.9086/0.746 = 1.218 m3/s

**Comment by rraustad** [comment]
Given that @larryscheier's math above exactly matches the results, then these changes accurately report simulation inputs/results. @mjwitte are there any more changes on the horizon?

**Comment by mjwitte** [comment]
@rraustad No more changes planned.

**Comment by Myoldmopar** [comment]
Inclined to merge.  Do we need to occupy time on the call in an hour to discuss this further?  I'd rather drop this in and talk about the other remaining items as it looks like we're all happy with this.  I'll merge if I don't hear anything before the call.

**Comment by rraustad** [comment]
I'm OK with that. I was going to run a few defect files tonight but there is plenty of other things to do.

**Comment by mjwitte** [comment]
No objection from me.

**Pull request merged**